lib/modules/datasource/puppet-forge/index.ts

```diff
@@ -0,0 +1,54 @@
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { PUPPET_FORGE } from './common';
+import type { PuppetModule } from './types';
+
+export class PuppetForgeDatasource extends Datasource {
+  static id = 'puppet-forge';
+
+  constructor() {
+    super(PuppetForgeDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = [PUPPET_FORGE];
+
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // https://forgeapi.puppet.com
+    const moduleSlug = packageName.replace('/', '-');
+    const url = `${registryUrl}/v3/modules/${moduleSlug}?exclude_fields=current_release`;
+
+    const { body: module } = await this.http.getJson<PuppetModule>(url);
```

Comment:
missing error handling, checkout other datasource

Reply:
there is an generic error handling in https://github.com/renovatebot/renovate/blob/037ed81544562e7a7f80c8bee4285b763dc3a95d/lib/modules/datasource/index.ts#L288
In my opinion I could not really do more than that.

Reply:
Add try catch and use `this.handleGenericErrors` in catch block